File: Historical PowerTrack/C#/MonitorJobStatus.cs
using System;
using System.Net;
using System.IO;
using System.Text;

namespace BasicOps
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			MainClass op = new MainClass();
			op.MonitorJob();
		}

		public HttpWebRequest makeRequest()
		{

            // Enter the jobURL for the historical job below.  This is a field returned when you create the job.
			string urlString = "ENTER_JOB_URL_HERE";

			string username = "ENTER_USERNAME";
			string password = "ENTER_PASSWORD";

			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(urlString);
		
			string authInfo = string.Format("{0}:{1}", username, password);
                        authInfo = Convert.ToBase64String(Encoding.Default.GetBytes(authInfo));
                        request.Headers.Add("Authorization", "Basic " + authInfo);
                
                        //In some Windows environments, this alternative Basic Authentication method is available.
                        //NetworkCredential nc = new NetworkCredential(username, password);
                        //request.Credentials = nc;
                        //request.PreAuthenticate = true;

			return request;			
		}


		public void MonitorJob()
		{
			HttpWebRequest request = makeRequest();
            request.Method = "GET";
			HttpWebResponse response = (HttpWebResponse) request.GetResponse();
            
			Console.WriteLine (((HttpWebResponse)response).StatusDescription);
			StreamReader reader = new StreamReader(response.GetResponseStream());

            string responseFromServer = reader.ReadToEnd ();
            Console.WriteLine (responseFromServer);
			Console.WriteLine();
            reader.Close ();
	        response.Close ();
		}

	}
}

File: Historical PowerTrack/Python/MonitorJobStatus.py
#!/usr/bin/env python
import urllib2
import base64


class RequestWithMethod(urllib2.Request):
    def __init__(self, url, method, headers={}):
        self._method = method
        urllib2.Request.__init__(self, url, headers)

    def get_method(self):
        if self._method:
            return self._method
        else:
            return urllib2.Request.get_method(self)

if __name__ == "__main__":
    url = 'ENTER_HISTORICAL_JOB_URL'
    username = 'ENTER_USERNAME'
    password = 'ENTER_PASSWORD'

    base64string = base64.encodestring("{0}:{1}".format(username, password))

    req = RequestWithMethod(url, 'GET')
    req.add_header('Content-type', 'application/json')
    req.add_header("Authorization", "Basic {}".format(base64string.strip()))

    try:
        response = urllib2.urlopen(req)
        print(response.read())
    except urllib2.HTTPError as e:
        print(e)
